proxy/http2/Http2ConnectionState.h

```diff
@@ -38,7 +38,7 @@ enum Http2SendADataFrameResult {
   HTTP2_SEND_A_DATA_FRAME_DONE       = 3,
 };
 
-enum Http2ShutdownState { NOT_INITIATED, INITIATED, IN_PROGRESS };
+enum Http2ShutdownState { NOT_PLANNED, NOT_INITIATED, INITIATED, IN_PROGRESS };
```

Comment:
I just don't understand how this `schedule_in` is safe. If the object is destroyed before the 2sec timeout, the event will run on a freed object AFAICT. I think you have to make it cancelable and cancel the event when Http2ConnectionState dies.

The `HTTP2_SESSION_EVENT_RECV` is different, since `send_connection_event()` delivers the event directly, so it is reentrant on the event handler.